tests/sles4sap/redirection_tests/ensa2_hanadb_refresh_sapinstances.pm

```diff
@@ -0,0 +1,88 @@
+# SUSE's openQA tests
+#
+# Copyright SUSE LLC
+# SPDX-License-Identifier: FSFAP
+# Maintainer: QE-SAP <qe-sap@suse.de>
+# Summary: Test module tests ENSA2 Central Services with HANA DB - refresh sapinstance ASCS & ERS.
+#   After refresh no failover happens, fail count stays at 0, cluster is healthy.
+
+use parent 'sles4sap::sap_deployment_automation_framework::basetest';
+
+use warnings;
+use strict;
+use testapi;
+use serial_terminal qw(select_serial_terminal);
+use sles4sap::console_redirection;
+use hacluster qw(check_cluster_state crm_check_resource_location);
+use sles4sap::sap_deployment_automation_framework::deployment qw(load_os_env_variables get_sdaf_instance_id);
+use saputils;
+
+sub execute_crm_resource_refresh_and_check {
+    my (%args) = @_;
+    my $instance_type = $args{instance_type};
+    my $instance_id = $args{instance_id};
+    my $instance_hostname = $args{instance_hostname};
+    my $instance_sid = get_required_var('SAP_SID');
+    my $resource = "rsc_sap_${instance_sid}_$instance_type$instance_id";
+
+    # Delete resource's recorded failures before refresh
+    record_info("Delete failcount", "delete sapinstance recorded failures of $resource");
+    assert_script_run("sudo crm_failcount --delete -r $resource -N $instance_hostname");
+    # Refresh resource
+    record_info("Refresh $instance_type", "refresh sapinstance $resource");
+    assert_script_run("sudo crm resource refresh $resource");
+
+    # Query the current value of the resource's fail count
+    record_info("Query $instance_type", "Query fail count of sapinstance $resource");
+    my $str = script_output("sudo crm_failcount --query -r $resource -N $instance_hostname");
+    my $value =~ /value=(\d+)/;
+    die 'Test failed to crm_failcount is non-zero' if ($value != '0');
+    # Check cluster
+    record_info('Cluster check', 'Checking state of cluster resources is healthy');
+    check_cluster_state();
+    # Check failover
+    record_info('NoFailover check', 'Checking no failover happens');
+    crm_check_resource_location(resource => $resource, wait_for_target => $instance_hostname);
+}
+
+sub run {
+    my ($self, $run_args) = @_;
+    my %redirection_data = %{$run_args->{redirection_data}};
+
+    # Connect to deployer VM (by default) to load env variables and get get instance IDs
+    connect_target_to_serial();
+    load_os_env_variables();
+    my $instance_id_scs = get_sdaf_instance_id(pattern => 'SCS');
```

Comment:
If possible, I would like to avoid having deployment specific functions from redirection tests. 
This will get us possibility to use this module in other deployment types as well. 

I am already working on lib functions for that purpose:
https://github.com/os-autoinst/os-autoinst-distri-opensuse/compare/master...lpalovsky:os-autoinst-distri-opensuse:ensa2_kill_enq_process#diff-4b80cb4a8bb93d5ab274aa60ee9f54d788f1ea66b09c11d416fe1de5e11befd4R103

So technically we can keep your code and I will change it later.

Reply:
> If possible, I would like to avoid having deployment specific functions from redirection tests. This will get us possibility to use this module in other deployment types as well.
> 
> I am already working on lib functions for that purpose: [master...lpalovsky:os-autoinst-distri-opensuse:ensa2_kill_enq_process#diff-4b80cb4a8bb93d5ab274aa60ee9f54d788f1ea66b09c11d416fe1de5e11befd4R103](https://github.com/os-autoinst/os-autoinst-distri-opensuse/compare/master...lpalovsky:os-autoinst-distri-opensuse:ensa2_kill_enq_process#diff-4b80cb4a8bb93d5ab274aa60ee9f54d788f1ea66b09c11d416fe1de5e11befd4R103)
> 
> So technically we can keep your code and I will change it later.

Got it, I will update the code when this lib function is available.